File: backend/src/main/java/com/horain/dev/DevSeedController.java
package com.horain.dev;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDate;

/**
 * Dev-only endpoint to load fictional seed data.
 * Disabled when horain.dev.seed-enabled is false (e.g. production).
 * Optional body: { "fixedToday": "2025-03-10" } for deterministic evals.
 */
@RestController
@RequestMapping("/dev")
public class DevSeedController {

    private final DevSeedService devSeedService;

    @Value("${horain.dev.seed-enabled:false}")
    private boolean seedEnabled;

    public DevSeedController(DevSeedService devSeedService) {
        this.devSeedService = devSeedService;
    }

    @PostMapping("/seed")
    public ResponseEntity<?> loadSeed(@RequestBody(required = false) SeedRequest body) {
        if (!seedEnabled) {
            return ResponseEntity.notFound().build();
        }
        LocalDate fixedToday = null;
        if (body != null && body.fixedToday() != null && !body.fixedToday().isBlank()) {
            try {
                fixedToday = LocalDate.parse(body.fixedToday());
            } catch (Exception ignored) {
                // Ignore invalid date, use now
            }
        }
        DevSeedService.DevSeedResult result = devSeedService.loadSeed(fixedToday);
        return ResponseEntity.ok(result);
    }

    @PostMapping("/seed/reset")
    public ResponseEntity<?> resetSeed(@RequestBody(required = false) SeedRequest body) {
        if (!seedEnabled) {
            return ResponseEntity.notFound().build();
        }
        LocalDate fixedToday = null;
        if (body != null && body.fixedToday() != null && !body.fixedToday().isBlank()) {
            try {
                fixedToday = LocalDate.parse(body.fixedToday());
            } catch (Exception ignored) {
                // Ignore invalid date, use now
            }
        }
        DevSeedService.DevSeedResult result = devSeedService.resetAndLoadSeed(fixedToday);
        return ResponseEntity.ok(result);
    }

    public record SeedRequest(String fixedToday) {}
}

File: backend/src/main/kotlin/com/horain/dev/DevSeedController.kt
package com.horain.dev

import org.springframework.beans.factory.annotation.Value
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDate

/**
 * Dev-only endpoint to load fictional seed data.
 * Disabled when horain.dev.seed-enabled is false (e.g. production).
 * Optional body: { "fixedToday": "2025-03-10" } for deterministic evals.
 */
@RestController
@RequestMapping("/dev")
class DevSeedController(
    private val devSeedService: DevSeedService
) {

    @Value("\${horain.dev.seed-enabled:false}")
    private var seedEnabled: Boolean = false

    @PostMapping("/seed")
    fun loadSeed(@RequestBody(required = false) body: SeedRequest?): ResponseEntity<*> {
        if (!seedEnabled) {
            return ResponseEntity.notFound().build<Any>()
        }
        val fixedToday = parseFixedToday(body?.fixedToday)
        val result = devSeedService.loadSeed(fixedToday)
        return ResponseEntity.ok(result)
    }

    @PostMapping("/seed/reset")
    fun resetSeed(@RequestBody(required = false) body: SeedRequest?): ResponseEntity<*> {
        if (!seedEnabled) {
            return ResponseEntity.notFound().build<Any>()
        }
        val fixedToday = parseFixedToday(body?.fixedToday)
        val result = devSeedService.resetAndLoadSeed(fixedToday)
        return ResponseEntity.ok(result)
    }

    private fun parseFixedToday(fixedToday: String?): LocalDate? {
        if (fixedToday.isNullOrBlank()) return null
        return try {
            LocalDate.parse(fixedToday)
        } catch (_: Exception) {
            null
        }
    }

    data class SeedRequest(val fixedToday: String?)
}
